src/cli.rs

```diff
@@ -22,3 +29,39 @@ impl Rover {
         }
     }
 }
+
+impl Report for Rover {
+    fn serialize_command(&self) -> Result<String, SputnikError> {
+        Ok(serde_json::to_string(&self)?)
+    }
+
+    fn is_enabled(&self) -> bool {
+        option_env!("APOLLO_TELEMETRY_DISABLED").is_none()
+    }
+
+    fn endpoint(&self) -> Result<Url, SputnikError> {
+        if let Some(url) = option_env!("APOLLO_TELEMETRY_URL") {
+            Ok(url.parse()?)
+        } else if cfg!(debug_assertions) {
```

Comment:
important to note here that when we're developing on our own machines that metrics will not be reported. want to make sure that's ok behavior